transport-native-kqueue/src/main/java/io/netty/channel/kqueue/KQueue.java

```diff
@@ -46,8 +49,29 @@
         if (cause != null) {
             UNAVAILABILITY_CAUSE = cause;
         } else {
-            UNAVAILABILITY_CAUSE = PlatformDependent.hasUnsafe() ? null :
+            cause = PlatformDependent.hasUnsafe() ? null :
                     new IllegalStateException("sun.misc.Unsafe not available");
+            if (cause == null) {
+                if (SystemPropertyUtil.get("os.name", "").toLowerCase(Locale.US).contains("mac")) {
+                    String version = SystemPropertyUtil.get("os.version", "");
+                    String[] versionParts = SystemPropertyUtil.get("os.version", "")
+                            .toLowerCase(Locale.US).split("\\.");
+                    if (versionParts.length < 2) {
+                        cause = new IllegalStateException("MacOS version " + version + " not supported");
+                    } else {
+                        try {
+                            int majorVersion = Integer.parseInt(versionParts[0]);
+                            int minorVersion = Integer.parseInt(versionParts[1]);
+                            if (majorVersion < 10 || (majorVersion == 10 && minorVersion < 12)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 5 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS134)